### ISSUE TITLE
Error Occurred During Phi-1.5 Fine-Tuning Process

### ISSUE THREAD

**Issue by then-go** [locuslab/open-unlearning#49]
Dear Authors,

While fine-tuning phi-1.5, I encountered the following issue:
	•	The loss suddenly drops to zero
	•	The grad_norm value becomes NaN
Error log is below.
``` W0208 22:07:50.309000 481216 site-packages/torch/distributed/run.py:793] 
W0208 22:07:50.309000 481216 site-packages/torch/distributed/run.py:793] *****************************************
W0208 22:07:50.309000 481216 site-packages/torch/distributed/run.py:793] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
W0208 22:07:50.309000 481216 site-packages/torch/distributed/run.py:793] *****************************************
[2025-02-08 22:07:55,119] [INFO] [real_accelerator.py:219:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2025-02-08 22:07:55,119] [INFO] [real_accelerator.py:219:get_accelerator] Setting ds_accelerator to cuda (auto detect)
Warning: The cache directory for DeepSpeed Triton autotune, /home/g1/.triton/autotune, appears to be on an NFS system. While this is generally acceptable, if you experience slowdowns or hanging when DeepSpeed exits, it is recommended to set the TRITON_CACHE_DIR environment variable to a non-NFS path.
Warning: The cache directory for DeepSpeed Triton autotune, /home/g1/.triton/autotune, appears to be on an NFS system. While this is generally acceptable, if you experience slowdowns or hanging when DeepSpeed exits, it is recommended to set the TRITON_CACHE_DIR environment variable to a non-NFS path.
num_devices: 2
max_steps: 625
/home/g1/.conda/envs/tofu/lib/python3.10/site-packages/transformers/training_args.py:1545: FutureWarning: `evaluation_strategy` is deprecated and will be removed in version 4.46 of 🤗 Transformers. Use `eval_strategy` instead
  warnings.warn(
[2025-02-08 22:08:02,187] [INFO] [comm.py:652:init_distributed] cdb=None
num_devices: 2
max_steps: 625
/home/g1/.conda/envs/tofu/lib/python3.10/site-packages/transformers/training_args.py:1545: FutureWarning: `evaluation_strategy` is deprecated and will be removed in version 4.46 of 🤗 Transformers. Use `eval_strategy` instead
  warnings.warn(
[2025-02-08 22:08:02,217] [INFO] [comm.py:652:init_distributed] cdb=None
[2025-02-08 22:08:02,218] [INFO] [comm.py:683:init_distributed] Initializing TorchBackend in DeepSpeed with backend nccl
Using the `WANDB_DISABLED` environment variable is deprecated and will be removed in v5. Use the --report_to flag to control the integrations used for logging result (for instance --report_to none).
Using the `WANDB_DISABLED` environment variable is deprecated and will be removed in v5. Use the --report_to flag to control the integrations used for logging result (for instance --report_to none).
[2025-02-08 22:08:02,650] [INFO] [config.py:733:__init__] Config mesh_device None world_size = 2
[2025-02-08 22:08:02,650] [INFO] [config.py:733:__init__] Config mesh_device None world_size = 2
[2025-02-08 22:08:03,350] [INFO] [partition_parameters.py:348:__exit__] finished initializing model - num_params = 341, num_elems = 1.42B
max_steps is given, it will override any value given in num_train_epochs
[2025-02-08 22:08:08,526][accelerate.utils.other][WARNING] - Detected kernel version 5.4.0, which is below the recommended minimum of 5.5.0; this can cause the process to hang. It is recommended to upgrade the kernel to the minimum version or higher.
max_steps is given, it will override any value given in num_train_epochs
[2025-02-08 22:08:08,814] [WARNING] [engine.py:1232:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****
Parameter Offload: Total persistent parameters: 544768 in 194 params

  0%|          | 0/625 [00:00<?, ?it/s]/home/g1/.conda/envs/tofu/lib/python3.10/site-packages/torch/autograd/graph.py:825: UserWarning: cuDNN SDPA backward got grad_output.strides() != output.strides(), attempting to materialize a grad_output with matching strides... (Triggered internally at /opt/conda/conda-bld/pytorch_1727971112454/work/aten/src/ATen/native/cudnn/MHA.cpp:674.)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
/home/g1/.conda/envs/tofu/lib/python3.10/site-packages/torch/autograd/graph.py:825: UserWarning: cuDNN SDPA backward got grad_output.strides() != output.strides(), attempting to materialize a grad_output with matching strides... (Triggered internally at /opt/conda/conda-bld/pytorch_1727971112454/work/aten/src/ATen/native/cudnn/MHA.cpp:674.)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
Could not estimate the number of tokens of the input, floating-point operations will not be computed
Could not estimate the number of tokens of the input, floating-point operations will not be computed

{'loss': 4512.374, 'grad_norm': nan, 'learning_rate': 4.960000000000001e-06, 'epoch': 0.25}
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 9.920000000000002e-06, 'epoch': 0.5}
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.4880000000000002e-05, 'epoch': 0.74}
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.4880000000000002e-05, 'epoch': 0.74}
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.88e-05, 'epoch': 1.24} 
```

However, when using the same script to fine-tune llama2-7b, the training proceeded without any issues like this.

```
'{'loss': 2.281, 'grad_norm': 12.540370499336115, 'learning_rate': 2.4800000000000004e-06, 'epoch': 0.25}
{'loss': 1.485, 'grad_norm': 4.456762068884375, 'learning_rate': 4.960000000000001e-06, 'epoch': 0.5}
...
```

 Since I was unable to identify the root cause even after debugging, I decided to post this issue here.

I would greatly appreciate it if you could help investigate this matter.
Below, I have included the scripts used for both phi-1.5 and llama2-7b fine-tuning for reference.

```
master_port=18765
split=full
model=phi
lr=2e-5


CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node=2 --master_port=$master_port finetune.py --config-name=finetune.yaml split=${split} batch_size=4 gradient_accumulation_steps=4 model_family=${model} lr=${lr}

master_port=18767
split=full
model=llama2-7b
lr=1e-5

CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node=2 --master_port=$master_port finetune.py --config-name=finetune.yaml split=${split} batch_size=4 gradient_accumulation_steps=4 model_family=${model} lr=${lr}
```

Additionally, I would like to ask about the weight_decay used in the fine-tuned models shared on Hugging Face:
	•	locuslab/tofu_ft_phi-1.5
	•	locuslab/tofu_ft_llama2-7b

When I evaluated these Hugging Face models against the results provided in the ./data folder, I noticed discrepancies. However, I could not determine the exact weight_decay value used during fine-tuning.

I would really appreciate any insights or clarification on this.
Thank you for your help!

**Comment by ruidazeng** [comment]
What is your output when you type `nvidia-smi` into the terminal?

**Comment by molereddy** [comment]
Marked closed due to inactivity, please note that the `locuslab` HF space is not being used anymore, as as it corresponds to the old version of TOFU, which is not being maintained anymore, see note on [https://github.com/locuslab/tofu](https://github.com/locuslab/tofu) 

Recently TOFU has been updated to  [https://github.com/locuslab/open-unlearning](https://github.com/locuslab/open-unlearning) which is a completely revamped framework using newer base model architectures and supporting more datasets.
You can find the new models at [https://huggingface.co/open-unlearning](https://huggingface.co/open-unlearning).

Please re-open if you face similar errors or reproducibilty issues with OpenUnlearning
